sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1053 @@
+     *        ISO-8859-1 character set.
+     * @remark You can leverage SerializeDataLakePropertiesString to create
+     *         the properties string from an std::map.
+     */
+    std::string Properties;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeFileSystemClient::Delete
+   */
+  struct DeleteFileSystemOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeFileSystemClient::GetProperties
+   */
+  struct GetFileSystemPropertiesOptions : public SharedRequestOptions
+  {
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeFileSystemClient::SetProperties
+   */
+  struct SetFileSystemPropertiesOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief User-defined properties to be stored with the filesystem, in
+     *        the format of a comma-separated list of name and value pairs
+     *        "n1=v1, n2=v2, ...".
+     *        Note that the string may only contain ASCII characters in the
+     *        ISO-8859-1 character set.
+     * @remark You can leverage SerializeDataLakePropertiesString to create
+     *         the properties string from an std::map.
+     */
+    std::string Properties;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeFileSystemClient::ListPaths
+   */
+  struct ListPathsOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief Valid only when Hierarchical Namespace is enabled for the account.
+     *        If "true", the user identity values returned in the owner and group
+     *        fields of each list entry will be transformed from Azure Active Directory
+     *        Object IDs to User Principal Names. If "false", the values will be
+     *        returned as Azure Active Directory Object IDs. The default value is false.
+     *        Note that group and application Object IDs are not translated because they
+     *        do not have unique friendly names.
+     */
+    bool Upn = bool();
```

Comment:
```suggestion
    bool UserPrincipalName = bool();
```

Reply:
Again, why it is resolved?

Reply:
It is already fixed.

Reply:
no, it's not fixed yet.

Reply:
@katmsft There's one more conversation not resolved here.